firmware/application/external/app_manager/ui_app_manager.cpp

```diff
@@ -0,0 +1,273 @@
+                            ui::Color::light_grey(),
+                            &bitmap_icon_sdcard,
+                            [this, app_id = std::string(app.appCallName)](KeyEvent) {
+                                button_hide_unhide.focus();
+                            }});
+    });
+
+    menu_view.set_highlighted(previous_cursor_index);
+}
+
+void AppManagerView::focus() {
+    menu_view.focus();
+}
+
+void AppManagerView::hide_app() {
+    std::vector<std::string> blacklist;
+    get_blacklist(blacklist);
+
+    blacklist.push_back(get_app_info(menu_view.highlighted_index(), true));
+    write_blacklist(blacklist);
+}
+
+void AppManagerView::unhide_app() {
+    std::vector<std::string> blacklist;
+    get_blacklist(blacklist);
+    blacklist.erase(std::remove(blacklist.begin(), blacklist.end(), get_app_info(menu_view.highlighted_index(), true)), blacklist.end());
+    write_blacklist(blacklist);
+}
+
+void AppManagerView::hide_unhide_app() {
+    if (is_blacklisted(get_app_info(menu_view.highlighted_index(), true)))
+        unhide_app();
+    else
+        hide_app();
+}
+
+void AppManagerView::get_blacklist(std::vector<std::string>& blacklist) {
+    File f;
+
+    auto error = f.open(u"SETTINGS/blacklist");
+    if (error)
+        return;
+
+    auto reader = FileLineReader(f);
+    for (const auto& line : reader) {
+        if (line.length() == 0 || line[0] == '#')
+            continue;
+
+        blacklist.push_back(trim(line));
+    }
+}
+
+void AppManagerView::write_blacklist(const std::vector<std::string>& blacklist) {
+    File f;
+    auto error = f.create(u"SETTINGS/blacklist");
+    if (error)
+        return;
+
+    for (const auto& app_name : blacklist) {
+        auto line = app_name + "\n";
+        f.write(line.c_str(), line.length());
+    }
+}
+
+void AppManagerView::clean_blacklist() {
+    std::vector<std::string> blacklist = {};
+    write_blacklist(blacklist);
+}
+
+bool AppManagerView::is_blacklisted(const std::string& app_name) {
+    std::vector<std::string> blacklist;
+    get_blacklist(blacklist);
+    return std::find(blacklist.begin(), blacklist.end(), app_name) != blacklist.end();
+}
+
+void AppManagerView::set_auto_start() {
+    auto app_index = menu_view.highlighted_index();
+    if (app_index >= app_list_index) return;
+
+    auto id_aka_friendly_name = get_app_info(app_index, false);
+
+    autostart_app = id_aka_friendly_name;
+
+    refresh_list();
+}
+
+void AppManagerView::unset_auto_start() {
+    autostart_app = "";
+    refresh_list();
+}
+
+void AppManagerView::set_unset_autostart_app() {
+    auto app_index = menu_view.highlighted_index();
+    if (app_index >= app_list_index) return;
+
+    auto id_aka_friendly_name = get_app_info(app_index, false);
+
+    if (is_autostart_app(id_aka_friendly_name))
+        unset_auto_start();
+    else
+        set_auto_start();
+}
+
+bool AppManagerView::is_autostart_app(const char* id_aka_friendly_name) {
```

Comment:
this have to be overload once, to satisfy the external app loader and inner app loader

Reply:
Okay. 
On A Side Note, id_aka_friendly_name is quite the variable name X-D